src/Components/Users/UserProfile.tsx

```diff
@@ -117,6 +117,8 @@ export default function UserProfile() {
     isChecking: false,
     isUpdateAvailable: false,
   });
+  const [hasChanges, setHasChanges] = useState<boolean>(false);
+  const [isFieldLoaded, setIsFieldLoaded] = useState<boolean>(false);
```

Comment:
Don't see a reason for having two states for achieving this. Would it be possible to just have one state `isDirty`?

Reply:
Actually, an additional state is required to detect the transition from the initial empty state to the actual state (when the data is received from the API call). This transition triggers the handleFieldChange function for the first time, so it's necessary.

Reply:
you can simply reset the `isDirty` state to false whenever the initial data is loaded. And set isDirty to true whenever any of the fields are updated right?

Reply:
`dirty` and `setDirty`? 🤔

Reply:
After hours of debugging, I discovered that the `PhoneNumberFormField.tsx` component performs mobile number validation in the `useEffect` hook on load. This triggers the `handleFieldChange` function, which updates the `isDirty` state prematurely. To prevent this behavior, I added the following conditional block to the `handleFieldChange` method. Does this look okay?

```js
const isMobileNumberModified =
  (event.name === "phoneNumber" || event.name === "altPhoneNumber") &&
  (event.value == userData?.alt_phone_number || event.value == userData?.phone_number);

if (isMobileNumberModified) {
  console.log(states.form);
  setDirty(false);
} else {
  setDirty(true);
}
```

Reply:
Hey @rithviknishad if i remove that `useEffect` than there is no affect in functionality so shall I made a commit?

Reply:
Yup

Reply:
Hey @rithviknishad Latest Changes done please check the updates.